src/blockrelay/graphene_set.cpp

```diff
@@ -56,11 +56,11 @@ CGrapheneSet::CGrapheneSet(size_t _nReceiverUniverseItems,
 
     // Construct Bloom filter
     pSetFilter = new CBloomFilter(
-        nItems, fpr, insecure_rand.rand32(), BLOOM_UPDATE_ALL, true, std::numeric_limits<uint32_t>::max());
+    static_cast<unsigned int>(nItems), fpr, insecure_rand.rand32(), BLOOM_UPDATE_ALL, true, std::numeric_limits<uint32_t>::max());
     LOG(GRAPHENE, "fp rate: %f Num elements in bloom filter: %d\n", fpr, nItems);
 
     // Construct IBLT
-    uint64_t nIbltCells = std::max((int)IBLT_CELL_MINIMUM, (int)ceil(optSymDiff));
+    uint64_t nIbltCells = (uint64_t) std::max((int)IBLT_CELL_MINIMUM, (int)ceil(optSymDiff));
```

Comment:
Hmm, this looks like a roundabout way with too many casts. First casting to `int` and then to `uint64_t`. Might there be a better way to deal with this, just casting the necessary parts to the same size on the RHS? This goes for quite a few of the fixes below as well.